#### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты.
    
   a) Найти вероятность того, что все карты – крести.
   
   б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

Решение: 

A)Всего карт одной масти в колоде из 52 карт  = 52/4 = 13
Вероятность выпадения первой карты крести = 13/52 = 1/4.
Допустим, первая выпала крести. Тогда остается 12 крести из 51 карты  
Вероятность выпадения второй карты крести = 12/51 = 4/17.
и т.д., 3ей = 11/50, 4й = 10/49

Чтобы вычислить вероятность выпадения всех 4х карт крести, нужно умножить вероятности:

P = 1/4 * 4/17 * 11/50 * 10/49 = 11/4165 или: 0,2641%

In [1]:
P1 = 13/52
P2 = 12/51
P3 = 11/50
P4 = 10/49

P1 * P2 * P3 * P4

0.0026410564225690276

Другое решение:
    А = {все карты крести}

$$P(A) = \frac{C_{13}^4}{C_{52}^4}$$

$$C_{13}^4 = \frac{13!}{4!(13 - 4)!}$$

In [ ]:
13*12*11*10/(2*3*4)

In [15]:
13*11*5

715

$$C_{52}^4 = \frac{52!}{4!(52 - 4)!}$$

In [ ]:
52*51*50*49/(2*3*4)

In [17]:
13*17*25*49

270725

In [13]:
715/270725

0.0026410564225690276

In [2]:
from math import factorial

In [3]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [4]:
C413 = combinations(13,4)
C413

715

In [5]:
C452 = combinations(52,4)
C452

270725

In [6]:
C413/C452

0.0026410564225690276

Б) Число сочетаний, когда одна карта - туз:

$$C_{4}^1 = \frac{4!}{1!(4 - 1)!}$$

2 * 3 * 4 / (2 * 3) = 4

Остальные три карты выбираются из 48 карт — тузы не рассматриваются, так как они уже выбраны:

$$C_{48}^3 = \frac{48!}{3!(48 - 3)!}$$

48 * 47 * 46 / (2 * 3)

8 * 47 * 46 = 17296

В этом случае будет такое число сочетаний, когда из колоды, в которой 52 карты, выбрано четыре, одна из которых — туз:

4 * 17296 = 69184

Число сочетаний, когда из 52 карты выбрано четыре, 2 из которых — туз:

In [18]:
T = combinations(4,2)
T

6

In [22]:
O = combinations(48,4-2)
O

1128

In [20]:
T*O

6768

In [ ]:
Число сочетаний, когда из 52 карты выбрано четыре, 3 из которых — туз:

In [21]:
T = combinations(4,3)
T

4

In [23]:
O = combinations(48,4-3)
O

48

In [24]:
T*O

192

In [ ]:
Число сочетаний, когда из 52 карты выбрано четыре, 4 (все)  — тузы: 1 (единственное)

In [25]:
T = combinations(4,4)
T

1

In [26]:
O = combinations(48,4-4)
O

1

In [27]:
T*O

1

Складываем полученные числа сочетаний:

69184 + 6768 + 192 + 1 = 76145

Выше выяснили, что всего сочетаний 4х карт из колоды 52 карты = 270725

Т.е. вероятность, что из 4х карт будет хотя бы один туз = 76145/270725 = 0.2812632745405855

Другое решение (подсмотренное в гугле):

In [30]:
1 - combinations(48,4)/combinations(52,4)

0.2812632745405854


   #### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9.
    Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?


Т.к. порядок не важен, ищем количество комбинаций 3 из 9:

$$C_{9}^3 = \frac{9!}{3!(9 - 3)!}$$

9 * 8 * 7 / (2 *3)

3 * 4 * 7 = 84

Т.к. попытка одна, вероятность открытия двери = 1 / 84 = 0.0119

In [33]:
1/84

0.011904761904761904

    
    
#### 3. В ящике имеется 15 деталей, из которых 9 окрашены.

Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?


Вероятность того, что первая деталь окрашена = 9 / 15 = 3 / 5
Вторая = 8 / 14 = 4 / 7
Третья = 7 / 13

Вероятность того, что все три детали окрашены = (3 / 5) * (4 / 7) * (7 / 13) = 12 / 65 = 0.1846

In [34]:
12/65

0.18461538461538463

#### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?


Всего комбинаций 2х билетов из ста:

$$C_{100}^2 = \frac{100!}{2!(100 - 2)!}$$

100 * 99 / 2 = 50 * 99 = 4950

Так как существует есдинственное сочетание двух выигрышных билетов из двух имеющихся, то вероятность того, что 2 приобретенных билета окажутся выигрышными, равна 1/ 4950 = 0,00(02)

In [35]:
1 / 4950

0.00020202020202020202